docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,499 @@
+
+A node (or nodes) can be brought back out of retirement using the `unretire`
+API:
+
+[source,js]
+--------------------------------------------------
+POST /_nodes/node_name/unretire
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) to reinstate are specified using <<cluster-nodes,node
+filters>> in place of `node_name` here. After being brought back out of
+retirement they may not immediately be added to the configuration.
+
+The current set of retired nodes is stored in the cluster state and can be
+inspected as follows:
+
+[source,js]
+--------------------------------------------------
+GET /_cluster/state?filter_path=TODO
+--------------------------------------------------
+// CONSOLE
+
+[float]
+=== Cluster bootstrapping
+
+There is a risk when starting up a brand-new cluster is that you accidentally
+form two separate clusters instead of one. This could lead to data loss: you
+might start using both clusters before noticing that anything had gone wrong,
+and it will then be impossible to merge them together later.
+
+NOTE: To illustrate how this could happen, imagine starting up a three-node
+cluster in which each node knows that it is going to be part of a three-node
+cluster. A majority of three nodes is two, so normally the first two nodes to
+discover each other will form a cluster and the third node will join them a
+short time later. However, imagine that four nodes were accidentally started
+instead of three: in this case there are enough nodes to form two separate
+clusters. Of course if each node is started manually then it's unlikely that
+too many nodes are started, but it's certainly possible to get into this
+situation if using a more automated orchestrator, particularly if a network
+partition happens at the wrong time.
+
+We avoid this by requiring a separate _cluster bootstrap_ process to take place
+on every brand-new cluster. This is only required the very first time the whole
+cluster starts up: new nodes joining an established cluster can safely obtain
+all the information they need from the elected master, and nodes that have
+previously been part of a cluster will have stored to disk all the information
+required when restarting.
+
+The simplest way to bootstrap a cluster is to use the
+`elasticsearch-bootstrap-cluster` command-line tool:
+
+[source,txt]
+--------------------------------------------------
+$ bin/elasticsearch-bootstrap-cluster --failure-tolerance 1 \
+    --node http://10.0.12.1:9200/ --node http://10.0.13.1:9200/ \
+    --node https://10.0.14.1:9200/
+--------------------------------------------------
+
+The arguments to this tool are the target failure tolerance of the cluster and
+the addresses of (some, preferably all, of) its master-eligible nodes.
+
+If it is not possible to use this tool, you can also bootstrap the cluster via
+the API as described here. There are two steps to the bootstrapping process.
```

Comment:
This is a bit misleading, because there is a way to just have one step if you configure node names.

Reply:
This is fixed now that the bootstrapping section is in a better order.